src/h264/h264.h

```diff
@@ -180,8 +226,328 @@ class H264SeqParameterSet
+			vcl_hrd_parameters_present_flag = r.Get(1);
+			if (vcl_hrd_parameters_present_flag) 
+			{
+				// vcl hrd_parameters()
+				if (!vcl_hrd_parameters.ParseHRDParams(r))
+					return false;
+			}
+
+			if (nal_hrd_parameters_present_flag || vcl_hrd_parameters_present_flag)
+			{
+				low_delay_hrd_flag = r.Get(1);
+			}
+
+			pic_struct_present_flag = r.Get(1);
+			bitstream_restriction_flag = r.Get(1);
+
+			if (bitstream_restriction_flag)
+			{
+				motion_vectors_over_pic_boundaries_flag = r.Get(1);
+
+				max_bytes_per_pic_denom = r.GetExpGolomb();
+				if (max_bytes_per_pic_denom < MaxBytesPerPicDenomMin ||
+					max_bytes_per_pic_denom > MaxBytesPerPicDenomMax) 
+				{
+					Warning("-H264SeqParameterSet::DecodeVuiParameters() Invalid max_bytes_per_pic_denom %u. Not in Range [%u, %u]\n", 
+							max_bytes_per_pic_denom, MaxBytesPerPicDenomMin, MaxBytesPerPicDenomMax);
+					return false;
+				}
+
+				max_bits_per_mb_denom = r.GetExpGolomb();
+				if (max_bits_per_mb_denom < MaxBitsPerMbDenomMin ||
+					max_bits_per_mb_denom > MaxBitsPerMbDenomMax) 
+				{
+					Warning("-H264SeqParameterSet::DecodeVuiParameters() Invalid max_bits_per_mb_denom %u. Not in Range [%u, %u]\n", 
+							max_bits_per_mb_denom, MaxBitsPerMbDenomMin, MaxBitsPerMbDenomMax);
+					return false;
+				}
+
+				log2_max_mv_length_horizontal = r.GetExpGolomb();
+				if (log2_max_mv_length_horizontal < Log2MaxMvLengthHorizontalMin ||
+					log2_max_mv_length_horizontal > Log2MaxMvLengthHorizontalMax) 
+				{
+					Warning("-H264SeqParameterSet::DecodeVuiParameters() Invalid log2_max_mv_length_horizontal %u. Not in Range [%u, %u]\n", 
+							log2_max_mv_length_horizontal, Log2MaxMvLengthHorizontalMin, Log2MaxMvLengthHorizontalMax);
+					return false;
+				}
+
+				log2_max_mv_length_vertical = r.GetExpGolomb();
+				if (log2_max_mv_length_vertical < Log2MaxMvLengthVerticalMin ||
+					log2_max_mv_length_vertical > Log2MaxMvLengthVerticalMax) 
+				{
+					Warning("-H264SeqParameterSet::DecodeVuiParameters() Invalid log2_max_mv_length_vertical %u. Not in Range [%u, %u]\n", 
+							log2_max_mv_length_vertical, Log2MaxMvLengthVerticalMin, Log2MaxMvLengthVerticalMax);
+					return false;
+				}
+
+				max_num_reorder_frames = r.GetExpGolomb();
+				if (max_num_reorder_frames < 0 ||
+					max_num_reorder_frames > MaxDpbFrames) 
+				{
+					Warning("-H264SeqParameterSet::DecodeVuiParameters() Invalid max_num_reorder_frames %u. Not in Range [%u, %u]\n", 
+							max_num_reorder_frames, 0, MaxDpbFrames);
+					return false;
+				}
+
+				max_dec_frame_buffering = r.GetExpGolomb();
+				if (max_dec_frame_buffering < 0 ||
+					max_dec_frame_buffering > MaxDpbFrames)
+				{
+					Warning("-H264SeqParameterSet::DecodeVuiParameters() Invalid max_dec_frame_buffering %u. Not in Range [%u, %u]\n", 
+							max_dec_frame_buffering, 0, MaxDpbFrames);
+					return false;
+				}
+			}
+			
+			return true;
+			
+		}
+		void DumpVuiParams() const
```

Comment:
keep naming consistency
```suggestion
		void Dump() const
```